torch/_logging/_internal.py

```diff
@@ -1063,6 +1066,10 @@ def trace_structured(
                 record["frame_id"] = trace_id.compile_id.frame_id
                 record["frame_compile_id"] = trace_id.compile_id.frame_compile_id
                 record["attempt"] = trace_id.attempt
+            else:
+                # Record the stack of the log call to better diagnose why we
+                # don't have a frame id for it
+                record["stack"] = torch._logging.structured.from_traceback(CapturedTraceback.extract(skip=1).summary())
```

Comment:
OOC, how do we reconstruct the stack for the trie when the compile id is known? Is there some mapping we get from the frame_id?

Reply:
When we start compilation, we dump the user stack trace of the code that triggered the compile. Those stacks get put together.